sciplot/Figure.hpp

```diff
@@ -38,10 +38,10 @@ class Figure
 {
   public:
     /// Construct a Figure object with given plots.
-    Figure(const std::initializer_list<std::initializer_list<Plot>>& plots);
+    Figure(const std::initializer_list<std::initializer_list<PlotBase*>>& plots);
```

Comment:
OK, I now see you have `Plot` and `Plot3D` inheriting from `PlotBase` and why the need for the pointer here. 

I would like to avoid the need for users to use pointers. Can't we have everything in the `Plot` class? This would be the best solution possibly.

If not, we could use `std::variant` or `std::any` here.

~~~c++
using PlotXD = std::variant<Plot, Plot3D>;
~~~

Reply:
I also first thought about how best to do it. Everything in a plot class I then found silly, because most methods in the 3d case are then not usable. Which is kind of confusing for the user. The pointers were just a first try to see how it looks like. I didn't want to put too much work into it, if it might be rejected.

Reply:
Good point. A `Plot3D` class is fine, but we should avoid constructs that will cause users to use pointers directly.

Reply:
Could you elaborate what functions are shared between 2D and 3D, and which are not?

Reply:
> ```c++
> using PlotXD = std::variant<Plot, Plot3D>;
> ```
Kind of like your std::variant idea too, Allan.

Reply:
I have not used the alias now.
It might be a bit more work to write, but at least you can see directly which types can be passed.
That is why I found variant also better than any

Reply:
2D and 3D currently share almost all format options (legend, font, size, palette) and also the basic functions for saving and displaying.